misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/evaluation/engine/ModuleLevelDefinitionFinder.java

```diff
@@ -98,12 +99,25 @@ protected void visitSyntaxNode(Node node) {
             return;
         }
 
-        // Need to ignore the entry points, when capturing top level definitions.
+        // Ignores entry points (main function definitions).
         if (node instanceof FunctionDefinitionNode && ((FunctionDefinitionNode) node).functionName().toSourceCode()
                 .equals(MAIN_FUNCTION_NAME)) {
             return;
         }
 
+        // Ignores external function definitions.
+        if (node instanceof FunctionDefinitionNode && ((FunctionDefinitionNode) node).functionBody().kind() ==
+                SyntaxKind.EXTERNAL_FUNCTION_BODY) {
+            return;
+        }
+
+        // Ignores type definitions with @builtinSubtype annotations (specific to Ballerina library sources).
+        if (node instanceof TypeDefinitionNode && ((TypeDefinitionNode) node).metadata().isPresent() &&
+                ((TypeDefinitionNode) node).metadata().get().annotations().stream().anyMatch(annotationNode ->
+                        annotationNode.annotReference().toSourceCode().trim().equalsIgnoreCase("builtinSubtype"))) {
```

Comment:
Shall we move the hardcoded `builtinSubtype` string to a constant?

Reply:
Done